src/fec_base.h

```diff
@@ -782,6 +783,9 @@ void FecCode<T>::decode_vec_lagrange(
     if (this->fft == nullptr) {
         throw LogicError("FEC base: FFT must be initialized");
     }
+    if (this->fft_full == nullptr) {
```

Comment:
Hum, these checks begin to multiply a bit too much.

Maybe we should have another approach to be sure, by construction, that these case cannot happen.
I don't know if my suggestion about using an abstract accessor (in aother comment) is a good solution here as well, if not we can keep it like that for now, but I'll try to find a better way to handle that.